### PR TITLE
add save KBaseSearch GenomeSet

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

**Comment by Tianhao-Gu** [comment]
@briehl Do you mind taking a look at this PR? There is a battle on using new GenomeSet vs old GenomeSet. I added the ability to create old GenomeSet.

**Comment by briehl** [comment]
@Tianhao-Gu I think the code looks reasonable. Could you update the SetAPI.spec to show the new parameter options?

**Comment by Tianhao-Gu** [comment]
@briehl Sure. updated spec.

File: lib/SetAPI/genome/GenomeSetInterfaceV1.py [hunk 1]
```diff
@@ -24,18 +34,24 @@ def save_genome_set(self, ctx, params):
             'set_info': info
         }
 
-    def _validate_genome_set_data(self, data):
+    def _validate_genome_set_data(self, data, save_search_set):
         # TODO: add checks that only one copy of each genome data is in the set
+        if save_search_set:
+            if 'elements' not in data:
+                raise ValueError('"elements" list must be defined in data to save an KBaseSearch.GenomeSet')
```
Comment (briehl) [inline lib/SetAPI/genome/GenomeSetInterfaceV1.py]: ... to save a KBaseSearch.GenomeSet, not 'an'

Comment (briehl) [reply lib/SetAPI/genome/GenomeSetInterfaceV1.py]: And is the 'elements' part of the GenomeSet object documented anywhere? Could you add it as an option to the spec if not?

**Review by briehl — CHANGES_REQUESTED** [review]
A couple of doc changes

**Comment by Tianhao-Gu** [comment]
@briehl 👍 sounds good. Updated GenomeSet spec.

**Review by briehl — APPROVED** [review]

**Comment by briehl** [comment]
Great! I'll merge after tests.

**Comment by briehl** [comment]
Sorry, got distracted. Merged!

**Pull request merged**